libs/ui/core/src/lib/brain/error-state-tracker.ts

```diff
@@ -0,0 +1,31 @@
+import { signal } from "@angular/core";
+import type { AbstractControl, FormGroupDirective, NgControl, NgForm } from "@angular/forms";
+import type { ErrorStateMatcher } from "./error-options";
+
+export class ErrorStateTracker {
+	/** Whether the tracker is currently in an error state. */
+	errorState = signal(false);
```

Comment:
I don't love errorState as a name. I feel like I'd expect an object with more information instead of a boolean, maybe something like hasError?

Reply:
I can change it, any idea about the content of the object?
You mean something like

```
state = signal<{ hasError: boolean, ... }>({ hasError: false, ... })
```